lib/tasks/accounts/update_adopter_status.rake

```diff
@@ -0,0 +1,36 @@
+namespace :accounts do
+  desc 'Update newly created field for adopter status from Salesforce'
+  # rake accounts:update_adopter_status
+  task update_adopter_status: [:environment] do
+    last_id = nil
+    loop do
+      contacts = OpenStax::Salesforce::Remote::Contact.select(
+        :id,
+        :adoption_status,
+        :accounts_uuid
+      )
+      .where("Accounts_UUID__c != null")
+      .order(:Id)
+      .limit(250)
+      contacts = contacts.where("Id > '#{last_id}'") unless last_id.nil?
+      contacts = contacts.to_a
+      last_id = contacts.last&.id
+
+      begin
+        user_by_salesforce_id = User.where(
+          salesforce_contact_id: contacts.map(&:id)
+        ).index_by(&:salesforce_contact_id)
+
+        updated_contacts = contacts.map do |contact|
+          user = user_by_salesforce_id[contact.id]
+          user.adopter_status = contact.adoption_status
+          user.save!
+        end
+      rescue StandardError => se
```

Comment:
If this rescues an error, you'll skip up to 250 contact (the entire batch) which is not great. May be less of a problem if we make this resumable. See below.

Reply:
I think you can remove the rescue now. Just let it fail if something unexpected happens. Otherwise it can get stuck in an infinite loop.

If it fails, you can always retry.